doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,211 @@
+ due to vascular contributions
+
+In the following example we show how to fit the IVIM model
+on a diffusion weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using Dipy's data fetcher `read_ivim`.
+This dataset was acquired with 21 b-values in 3 different directions.
+Volumes corresponding to different directions were registered to each
+other, and averaged across directions. Thus, this dataset has 4 dimensions,
+with the length of the last dimension corresponding to the number
+of b-values.
+"""
+
+img, gtab = read_ivim()
+
+"""
+img contains a nibabel Nifti1Image object(with the data)
+and gtab contains a GradientTable object(information about
+the gradients e.g. b-values and b-vectors). We get the
+data from img using `read_data`.
+"""
+
+data = img.get_data()
+print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+"""
+The data has 54 slices, with 256-by-256 voxels in each slice.
+The fourth dimension corresponds to the b-values in the gtab.
+Let us visualize the data by taking a slice midway(z=27) at bvalue = 0.
+"""
+
+z = 33
+b = 20
+
+plt.imshow(data[:, :, z, b].T, origin='lower', cmap='gray')
+plt.axhline(y=100)
+plt.axvline(x=170)
+plt.savefig("ivim_data_slice.png")
+plt.close()
+
+"""
+The marked point in the image shows a section containing cerebral spinal fluid (CSF)
+so it should have a very high f and D*, the area between the right and left is white
+matter so that should be lower, and the region on the right is gray matter and CSF.
+That should give us some contrast to see the values varying across the regions.
+"""
+
+x1, x2 = 160, 180
+y1, y2 = 90, 110
+data_slice = data[x1:x2, y1:y2, z, :]
+
+plt.imshow(data[x1:x2, y1:y2, z, b].T, origin='lower', cmap="gray")
+plt.savefig("CSF_slice.png")
+plt.close()
+
+"""
+Now that we have prepared the datasets we can go forward with
+the ivim fit. Instead of fitting the entire volume, we focus on a
+small section of the slice, to fit the IVIM model. First, we instantiate
+the Ivim model. Here, we use a two-stage approach: first, a tensor
+is fit to the data, and then this tensor as the initial starting point
+for the non-linear fit of IVIM parameters. All initializations are
+passed with the `IvimModel` such as `x0` or `split_b`. If you are using
+Scipy 0.17, you can also set bounds by setting
+`bounds=([0., 0., 0., 0.], [np.inf, 1., 1., 1.]))` while initializing
+the IvimModel. It is recommeded that you upgrade to Scipy 0.17 since
+the fitting results might at times return values which do not make sense
+physically. (Negative f values or D*)
+"""
+
+ivimmodel = IvimModel(gtab, x0=[100., 0.1, 0.01, 0.01])
+
+"""
+To fit the model, call the `fit_method` and pass the data for fitting.
+"""
+
+ivimfit = ivimmodel.fit(data_slice)
```

Comment:
The problem here is that x0 that gets derived in the first step is out of the bounds. This might have to do with the fact that this voxel has diffusion-weighted signals that are *higher* than the non-diffusion (b0) signal. This might cause some confusion in the first step of fitting.

One way to deal with this is to introduce a rectification of x0 to the bounds, before sending it into the fitting procedure.

How does that sound?